xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecStarfish.cpp

```diff
@@ -556,3 +579,91 @@ void CDVDVideoCodecStarfish::PlayerCallback(const int32_t type,
 {
   static_cast<CDVDVideoCodecStarfish*>(data)->PlayerCallback(type, numValue, strValue);
 }
+
+std::unique_ptr<IStarfishResource> IStarfishResource::Create()
+{
+  if (StarfishResourceWayland::IsAvailable())
+    return std::make_unique<StarfishResourceWayland>();
+  return std::make_unique<StarfishResourceAcb>();
+}
+
+bool StarfishResourceWayland::UpdateLoadPayload(CVariant& payload)
+{
+  using namespace KODI::WINDOWING::WAYLAND;
+  auto winSystem = static_cast<CWinSystemWaylandWebOS*>(CServiceBroker::GetWinSystem());
+
+  std::string exportedWindowName = winSystem->GetExportedWindowName();
+  payload["args"][0]["option"]["windowId"] = exportedWindowName;
+  CLog::LogF(LOGDEBUG, "CDVDVideoCodecStarfish: Setting window ID {} for load payload",
+             exportedWindowName);
+  return true;
+}
+
+bool StarfishResourceWayland::IsAvailable()
+{
+  using namespace KODI::WINDOWING::WAYLAND;
+  auto winSystem = static_cast<CWinSystemWaylandWebOS*>(CServiceBroker::GetWinSystem());
+  return winSystem->SupportsExportedWindow();
+}
+
+StarfishResourceAcb::StarfishResourceAcb()
+{
+  acbId = AcbAPI_create();
+  if (acbId)
+  {
+    AcbAPI_initialize(acbId, PLAYER_TYPE_MSE, getenv("APPID"), AcbCallback);
+  }
+}
+
+StarfishResourceAcb::~StarfishResourceAcb()
+{
+  if (acbId)
+    AcbAPI_destroy(acbId);
+}
+
+bool StarfishResourceAcb::PostLoad(CDVDStreamInfo& hints)
+{
+  if (!acbId)
+    return false;
+  AcbAPI_setDisplayWindow(acbId, 0, 0, hints.width, hints.height, true, nullptr);
```

Comment:
Unfortunately, this has to happen in `CRendererStarfish`. I'd suggest adding a field for `acbId` in the `StarfishVideoBuffer`. Then calling the function in `ManageRenderArea`. Also, videos may not always run fullscreen. For instance, if you display a SBS 3D video as 2D. There is some logic inside the renderer that calculates the target display coordinates. There's another catch here though. The target coordinates can be negative which the API will not accept. You can test this when a video partly goes offscreen, i.e. when using a zoom option > 1.